parallaxinc/Parallax-IDE
# New file save as

#### What's this PR do?

Adds an action queueing mechanism to allow us to pause execution of a behavior (e.g. open a "Save As" dialog when trying to exit a new file with text) but continue it after some action has been taken in the UI.  We used to have a naive implementation that only took into account when a user tried to change to an existing file while having a new, unsaved file open.  I added logic to queue up when a file is being changed or a new file is being created and allowed the implementation to not care if other queued actions are taken.  One caveat is that only 1 action can be queued up at a time, which makes the implementation simpler and should be fine because we use this functionality for modal prompting scenarios.

I switched up the eslint stuff to my new pattern also.
#### What are the important parts of the code?

Adds a new reducer and creators to maintain the action queue.  Adds logic to `src/plugins/handlers.js` that utilizes the action queue to kick off functions based on that current state.  Also, fixed a problem where the `isNew` flag wasn't being reset properly inside the `changeFile` function.
#### How should this be tested by the reviewer?

Create a new file, enter some text, try creating another new file.  You should be prompted to save that file.  Use "Save As", "Don't Save" and "Cancel" buttons.  Try switching between a new file with text to a saved file and back and forth.  Try all of these with a new file that has no text in it.  Close the editor while a new file with text is present, open it and make sure it prompts when you try to create another new file.
#### Is any other information necessary to understand this?

I think everything is covered in the first wall of text.
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

Fixes #230
